pipelines/monitor_forecast_trigger.py

```diff
@@ -0,0 +1,341 @@
+SIMULATE_TRIGGER = load_boolean_env("SIMULATE_TRIGGER", False)
+DRY_RUN = load_boolean_env("DRY_RUN", True)
+
+TEST_FORECAST_BLOB_NAME = os.getenv("TEST_FORECAST_BLOB_NAME", "")
+
+YASA_TEST_BLOB_NAME = f"{PROJECT_PREFIX}/raw/fms/TC Data/TC Yasa/20201216T000000Z_Official_Forecast_Track_2021_02F_YASA.csv"
+
+INFO_LIST_IDS = TEST_LIST_IDS if TEST_EMAIL else PROD_INFO_LIST_IDS
+TRIGGER_LIST_IDS = TEST_LIST_IDS if TEST_EMAIL else PROD_TRIGGER_LIST_IDS
+
+# enable/disable tqdm progress bars for running locally
+DISABLE_TQDM = load_boolean_env("DISABLE_TQDM", True)
+
+if __name__ == "__main__":
+    # Init
+    logger.info("Monitor forecast trigger pipeline started.")
+    logger.info(
+        f"Env vars set to: {TEST_EMAIL=}, {SIMULATE_TRIGGER=}, {DRY_RUN=}"
+    )
+    args = parse_args()
+    csv_str = args.csv
+    if csv_str:
+        decoded_csv = decode_b64_string(csv_str)
+    elif SIMULATE_TRIGGER:
+        decoded_csv = BytesIO(stratus.load_blob_data(YASA_TEST_BLOB_NAME))
+    elif TEST_FORECAST_BLOB_NAME:
+        decoded_csv = BytesIO(stratus.load_blob_data(TEST_FORECAST_BLOB_NAME))
+    else:
+        logger.info("No forecast CSV provided and no test blob set; exiting.")
+        exit(0)
+
+    # Load forecast
+    gdf_forecast = parse_fms_forecast(decoded_csv)
+    gdf_forecast = gdf_forecast.rename(columns={"forecast_time": "valid_time"})
+
+    row = gdf_forecast.iloc[0]
+    issued_time = row["base_time"]
+    cyclone_name = row["cyclone_name"]
+    season = row["season"]
+    logger.info(f"Issue time: {issued_time}, Cyclone name: {cyclone_name}")
+    forecast_display_str = fji_time_str(issued_time)
+    forecast_id = (
+        f"{cyclone_name.lower().replace(' ', '_')}_{season}_fcast_"
+        f"{issued_time:%Y%m%dT%H%MZ}"
+    )
+
+    logger.info(f"Forecast ID: {forecast_id=}")
+
+    gdf_readiness = gdf_forecast[gdf_forecast["leadtime"] <= 120].copy()
+    gdf_action = gdf_forecast[gdf_forecast["leadtime"] <= 72].copy()
+
+    # Calculate single track buffers
+    logger.info("Calculating forecast buffers.")
+    gdf_buffers_readiness = calculate_fms_buffers_gdf(gdf_readiness)
+    gdf_buffers_action = calculate_fms_buffers_gdf(gdf_action)
+
+    # Calculate population exposure
+    logger.info("Calculating population exposure.")
+    adm3 = codab.load_codab_from_blob(admin_level=3).to_crs(FJI_CRS)
+    da_wp = worldpop.load_worldpop_from_blob()
+    da_wp = da_wp.assign_coords({"x": ((da_wp.x + 360) % 360)}).sortby("x")
+    da_wp_clip = da_wp.rio.clip(adm3.geometry)
+
+    df_exp_readiness = calculate_single_adm_exposure(
+        gdf_buffers_readiness, da_wp_clip
+    )
+    df_exp_action = calculate_single_adm_exposure(
+        gdf_buffers_action, da_wp_clip
+    )
+
+    # Check trigger conditions
+    readiness_exp = df_exp_readiness[
+        df_exp_readiness["buffer_speed"] == 64
+    ].iloc[0]["pop_exposed"]
+    action_exp = df_exp_action[df_exp_action["buffer_speed"] == 64].iloc[0][
+        "pop_exposed"
```

Comment:
edge case that could crash pipeline:
                                                                                                              
If all forecast points have leadtime > 72, then gdf_action is empty → df_exp_action has no rows → .iloc[0] raises IndexError.                                                                
                                                                                                                                                                                                         
scenario: : Early forecast when storm is far away (e.g., 5-day forecast with no points within 72h).